python/ray/tune/ray_trial_executor.py

```diff
@@ -212,16 +226,47 @@ def _setup_remote_runner(self, trial, reuse_allowed):
             logger.debug("Cannot reuse cached runner {} for new trial".format(
                 self._cached_actor))
             with self._change_working_directory(trial):
-                self._trial_cleanup.add(trial, actor=self._cached_actor)
+                self._trial_cleanup.add(
+                    trial,
+                    actor=self._cached_actor,
+                    placement_group=self._committed_placement_groups.get(
+                        trial, [None])[0])
+                if trial in self._committed_placement_groups:
+                    del self._committed_placement_groups[trial]
             self._cached_actor = None
 
         _actor_cls = _class_cache.get(trial.get_trainable_cls())
-        full_actor_class = _actor_cls.options(
-            num_cpus=trial.resources.cpu,
-            num_gpus=trial.resources.gpu,
-            memory=trial.resources.memory or None,
-            object_store_memory=trial.resources.object_store_memory or None,
-            resources=trial.resources.custom_resources)
+        if isinstance(trial.resources, PlacementGroupFactory):
+            if trial not in self._committed_placement_groups:
+                # This creates the placement group and will try to
+                # allocate resources
+                placement_group = trial.resources()
+                ready_fut = placement_group.ready()
+                self._committed_placement_groups[
+                    trial] = placement_group, ready_fut
+            else:
+                placement_group, ready_fut = self._committed_placement_groups[
+                    trial]
+            ready, _ = ray.wait([ready_fut], timeout=0)
+            if not ready:
+                return None
+            if not ray.get(ready):
+                # Schedule new future
+                self._committed_placement_groups[
+                    trial] = placement_group, placement_group.ready()
+                return None
```

Comment:
This block might be unneeded - is there ever the case that the actual return value of `ready()` is evaluated to `False`?